Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -1687,7 +1694,39 @@ private void LogCommandLineTooLong(SandboxedProcessInfo info)
                 allReportedFileAccesses: allFileAccesses,
                 detouringStatuses: result.DetouringStatuses,
                 maxDetoursHeapSize: maxDetoursHeapSize,
-                containerConfiguration: m_containerConfiguration);
+                containerConfiguration: m_containerConfiguration,
+                pipProperties: pipProperties);
+        }
+
+        private async Task UpdatePipPropertiesCountsAsync(SandboxedProcessResult result, Dictionary<string, int> pipProperties)

```

Comment:
>UpdatePipPropertiesCountsAsync [](start = 27, length = 30)

This is only called once per call to `ProcessSandboxedProcessResultAsync`. So I don't understand when the `pipProperties` gets "updated". In this case perhaps the code should be 
```
private async Task<Dictionary<string, int>> GetPipPropertiesCountAsync(...);
```

Reply:
It updates whatever value was set on the result, which could originally be set prior to a retry and passed through to the current execution.  You're correct that it won't be set prior to the first instance.